torch/nn/functional.py

```diff
@@ -4761,15 +4761,22 @@ def _in_projection_packed(
     if k is v:
         if q is k:
             # self-attention
-            return linear(q, w, b).chunk(3, dim=-1)
+            proj = linear(q, w, b)
+            # reshape to 3, E and not E, 3 is deliberate for better memory coalescing and keeping same order as chunk()
+            proj = proj.unflatten(-1, (3, E)).unsqueeze(0).transpose(0, -2).squeeze(-1).contiguous()
```

Comment:
`squeeze(-1)` doesn't look right, last dimension most likely isn't 1 (unless E happens to be 1).

Reply:
Should be `squeeze(-2)` probably...

Reply:
Oops yes you're right, forgot to change when I updated from `transpose(0, -1)`. Still works because of the view() later, but I'll fix